zrml/swaps/src/math.rs

```diff
@@ -199,10 +198,11 @@ pub fn calc_single_out_given_pool_in(
     total_weight: u128,
     pool_amount_in: u128,
     swap_fee: u128,
+    exit_fee: u128,
 ) -> Result<u128, DispatchError> {
```

Comment:
If we remove the swap fee here, can this be abused to get an advantage by using pool join / exits as an alternative to using the swap functionality for trade?

Reply:
I am also confused why an exit fee is used here, since this is function is only used for joining operations.

Reply:
> If we remove the swap fee here, can this be abused to get an advantage by using pool join / exits as an alternative to using the swap functionality for trade?

Haven't done the exact testing (I'll do that when swap fees are implemented), but I see this approach. Suppose you have x ZTG and want to buy asset A from the pool, and suppose the pool has the standard weights (1/n for outcome tokens, 1 for ZTG; currently, all pools created via the frontend have this weight). Then you do the following:

- Buy x/2 complete sets
- `pool_join` with exactly x/2 of each asset
- `pool_exit_with_exact_pool_amount` with the pool amount received in the second step and A as the asset

This will cost you transaction fees (constant) and exit fees. I expect you get approximately the same amount as when doing `swap_exact_amount_in` with `x` as the amount, but instead of paying swap fees, you just pay the exit fees.

> I am also confused why an exit fee is used here, since this is function is only used for joining operations.

It's actually used in `pool_exit_with_exact_pool_amount`. The function names are just super confusing here. ;)

Reply:
Oh my, indeed!
Okay, I assume the best approach would be to leave the swap fee for single asset exits in until we have more information in regards to possible attacks without it.